src/Shipment.php

```
@@ -0,0 +1,352 @@
+     * for all available options for the sender,destination and parcel params.
+     * @return \DOMDocument
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function transmitShipments(
+        array $manifest_address,
+        array $group_ids,
+        array $options = []
+    ) {
+        $this->verifyPostalCode($manifest_address);
+        $content = [
+            'group-ids' => [
+                'group-id' => $group_ids
+            ],
+            'requested-shipping-point' => $manifest_address['address-details']['postal-zip-code'],
+            'cpc-pickup-indicator' => true,
+            'detailed-manifests' => true,
+            'manifest-address' => $manifest_address,
+        ];
+
+        if (!empty($options['option_codes'])) {
+            $content['options']['option'] = $this->parseOptionCodes($options);
+        }
+
+        $xml = Array2XML::createXML('transmit-set', $content);
+        $envelope = $xml->documentElement;
+        $envelope->setAttribute('xmlns',
+            'http://www.canadapost.ca/ws/manifest-v8');
+        $payload = $xml->saveXML();
+
+        $response = $this->post(
+            "rs/{$this->customerNumber}/{$this->customerNumber}/shipment",
+            [
+                'Accept' => 'application/vnd.cpc.manifest-v8+xml',
+                'Content-Type' => 'application/vnd.cpc.manifest-v8+xml',
+            ],
+            $payload,
+            $options
+        );
+        return $response;
+
+    }
+
+    /**
+     * Get the manifest from Canada Post server.
+     *
+     * @param string $manifest_id
+     *   The manifest id.
+     * @param string $extra
+     *   Gets the specific details for a shipment.
+     * @param array $options
+     *   The options array.
+     *
+     * @return \DOMDocument|\Psr\Http\Message\StreamInterface
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function getManifest($manifest_id, $extra = '', array $options = [])
+    {
```

Same as above, is `extra` required? Should we throw an exception if not-supported value is passed?